Analytical_Engine/Query/Graph/FilterRelations.cs

```diff
@@ -45,6 +45,12 @@ public static partial class Query
         [Output("filtered relations", "Collection of IRelations filtered from the Graph.")]
         public static List<IRelation> FilterRelations(this Graph graph, Type typeFilter)
         {
+            if (graph == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot filter the relations of a null graph.");
+                return null;
```

Comment:
Why not return an empty collection to follow the pattern?
```suggestion
                return new List<IRelation>();
```